x/posts/client/rest/tx.go

```diff
@@ -44,7 +44,16 @@ func createPostHandler(cliCtx context.CLIContext) http.HandlerFunc {
 			return
 		}
 
-		msg := types.NewMsgCreatePost(req.Message, parentID, req.AllowsComments, req.Subspace, req.OptionalData, addr, req.CreationTime)
+		var msg types.MsgCreatePost
+		msg = types.NewMsgCreateTextPost(req.Message, parentID, req.AllowsComments, req.Subspace, req.OptionalData,
+			addr, req.CreationTime)
+
+		if len(req.Medias) != 0 {
+			if textMsg, ok := msg.(types.MsgCreateTextPost); ok {
```

Comment:
This check will always succeed. Can we replace it with something like the following? 

```golang
msg = types.NewMsgCreateMediaPost(MsgCreateTextPost(msg), req.Medias)
```